### PR TITLE
Upgrade to SonarQube 7.9

### LINKED ISSUE #32: Upgrade to SonarQube 7.9
Upgrade to latest SonarQube version 7.9

### PR DESCRIPTION
**Pull request by mbehlendorf**
* Upgrade to SonarQube 7.9.1 (resolves #32)
* Use Java 11
* Use sonar-cas-plugin 2.0.1
* Enhance security of temporary admin user

### CODE REVIEW

**Comment by robertauer** [comment]
https://github.com/cloudogu/sonar/blob/a3ac21ff147ef6cc9f08c03ca076ed90b33f55df/resources/post-upgrade.sh#L68
This may be too late, set this key in pre-upgrade script.

File: resources/pre-upgrade.sh [hunk 1]
```diff
@@ -10,65 +10,74 @@ DATABASE_USER=$(doguctl config -e sa-postgresql/username)
...
+
 function remove_temporary_admin_user() {
   TEMPORARY_ADMIN_USER=${1}
   sql "DELETE FROM users WHERE login='${TEMPORARY_ADMIN_USER}';"
 }
 
-echo "Running pre-upgrade script..."
-if [[ ${FROM_VERSION} == *"5.6.7"* ]] && [[ ${TO_VERSION} == *"6.7."* ]]; then
-  TEMPORARY_ADMIN_USER="admin"
+if [[ ${FROM_VERSION} == *"6"* ]] && [[ ${TO_VERSION} == *"7.9"* ]]; then
```
Comment (robertauer) [inline resources/pre-upgrade.sh]: The comparison of the FROM_VERSION should be more specific. As of now, it also applies to sonar v5.6.6-4, for example.

File: resources/post-upgrade.sh [hunk 2]
```diff
@@ -120,20 +114,17 @@ if [[ ${FROM_VERSION} == *"5.6.7"* ]] && [[ ${TO_VERSION} == *"6.7."* ]]; then
...
+    reinstall_plugins "${TEMPORARY_ADMIN_USER}" "${PW}"
 
-    # clear install_plugins key
-    doguctl config install_plugins "" >> /dev/null
+    doguctl config --remove install_plugins
   fi
 
-  # Do everything that needs to be done to get into a state that is equal to a successful first start
-  create_dogu_admin_and_deactivate_default_admin ${CURL_LOG_LEVEL}
-  set_successful_first_start_flag
```
Comment (robertauer) [inline resources/post-upgrade.sh]: Why are these lines not necessary any more? 🤔

File: resources/pre-upgrade.sh [hunk 3]
```diff
@@ -10,65 +10,74 @@ DATABASE_USER=$(doguctl config -e sa-postgresql/username)
 DATABASE_USER_PASSWORD=$(doguctl config -e sa-postgresql/password)
 DATABASE_DB=$(doguctl config -e sa-postgresql/database)
 
+echo "Running pre-upgrade script..."
+if [[ ${FROM_VERSION} == *"5"* ]]; then
```
Comment (robertauer) [inline resources/pre-upgrade.sh]: This comparison should be more specific, e.g. *"5.6."*

File: resources/upgrade-notification.sh [hunk 4]
```diff
@@ -6,12 +6,17 @@ set -o pipefail
 FROM_VERSION="${1}"
 TO_VERSION="${2}"
 
-if [[ ${FROM_VERSION} == *"5.6.6"* ]] && [[ ${TO_VERSION} == *"6.7."* ]]; then
-  echo "You are upgrading from SonarQube 5.6.6 to a 6.7 LTS version. This may lead to unexpected behavior!"
-  echo "Please upgrade to 5.6.7 LTS before upgrading to 6.7 LTS!"
-  echo "See https://docs.sonarqube.org/latest/setup/upgrading/"
+if [[ ${FROM_VERSION} == *"5"* ]]; then
```
Comment (robertauer) [inline resources/upgrade-notification.sh]: This comparison should be more specific, e.g. *"5.6."*

File: resources/post-upgrade.sh (hunk 2 above)
Comment (mbehlendorf) [reply resources/post-upgrade.sh]: We needed these lines in cases where we upgraded from an old version, which didn't set the "successful_first_start_flag" nor created the dogu admin. We only execute this post_upgrade.sh if we come from a SonarQube 6, which already does this.

**Pull request merged**